scheduler/scheduler.cpp

```diff
@@ -624,72 +626,104 @@ static string envs_match(CompileServer *cs, const Job *job)
     return string();
 }
 
-static CompileServer *pick_server(Job *job)
+static list<CompileServer *> filter_ineligible_servers(Job *job)
 {
+    list<CompileServer *> eligible;
+    std::copy_if(
+        eligible.begin(),
```

Comment:
didn't this list just get created and so therefore its empty? Shouldn't you be iterating some populated list and back inserting into this new empty list?

Reply:
should you be iterating css? That seems like some kind of global list just reading the diffs, not sure where its coming from though.

Reply:
Yeah, I changed how this worked when I realized I didn't have what I wanted because this is C++11. I was trying to get it to compile and forgot to go back and change what I was iterating. It should be `css`, which is a static declared at the top of the file.